src/app.ts

```diff
@@ -6,91 +6,55 @@
  * (Department of Information and Computing Sciences)
  */
 
-import express, { NextFunction } from "express";
-import { Request, Response } from "express";
+import express, { Express, Request, Response, NextFunction } from "express";
+import cors from "cors";
+
 import { Context } from "./context";
-import { installRoute, uninstallRoute } from "./routes/install";
-import { body, param, query } from "express-validator";
-import { asyncCatch } from "./utils";
-import { handleValidationResult } from "./middlewares/validation";
+import { expressHandler } from "./utils";
+
+import { installHandler, uninstallHandler } from "./routes/install";
+import { AmqpSocket, createAmqpSocket } from "./amqp";
+import { createRoutingKeyStore } from "./routingKeyStore";
 import {
-  getAddonByIdRoute,
-  getAddonReadMeByIdRoute,
-  getAddonsRoute
+  getAddonByIdHandler,
+  getAddonReadMeByIdHandler,
+  getAddonsHandler
 } from "./routes/addons";
-import cors from "cors";
-import { AddonCategory } from "prisma/prisma-client";
 
-export function buildApp(ctx: Context) {
+export class App {
+  public constructor(
+    public express: express.Express,
+    public amqp: AmqpSocket
+  ) {}
+
+  public listen(port: number): void {
+    this.express.listen(port, () => {
+      console.log(`Server running on http://localhost:${port}`);
+    });
+
+    this.amqp.listen();
+    console.log("Listening for amqp messages");
+  }
+}
+
+export function buildExpress(ctx: Context): Express {
   const app = express();
   app.use(express.json());
 
-  app.use(
-    cors({
-      origin: "http://localhost:4201"
-    })
-  );
+  app.use(cors());
```

Comment:
My frontend port is different from the one listed here, so i got lazy and did this. We need to make a system to differentiate between dev mode and prod mode eventually and this needs to be wrapped in an if statement then.